src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutPersisterReceiver.cs

```diff
@@ -19,6 +19,16 @@ class TimeoutPersisterReceiver : IDisposable
         public CriticalError CriticalError { get; set; }
         public Address DispatcherAddress { get; set; }
         public TimeSpan TimeToWaitBeforeTriggeringCriticalError { get; set; }
+        internal DateTime NextRetrieval { get; private set; } = DateTime.UtcNow;
```

Comment:
Doesn't this set `NextRetrieval` before the instance of `TimeoutPersisterReceiver` is fully initialized? https://github.com/dotnet/roslyn/wiki/New-Language-Features-in-C%23-6#initializers-for-auto-properties

> Just like field initializers, auto-property initializers cannot reference `this` – after all they are executed before the object is properly initialized.

Does that cause a problem with the value being set too early and having an abnormally early value?

The way that this was implemented in `InMemoryTimeoutPersister` (as a `Func`) would preclude the problem of early initialization